### PR TITLE
Refactor analytics

### PR DESCRIPTION
**Pull request by irlaec**

### CODE REVIEW

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 1]
```diff
@@ -42,900 +40,635 @@ def existing_analyticsHandler():
...
+		self._cleanup_job()
+		self._current_job_id = 'j_{}_{}'.format(self._snr, time.time())
+		self._add_job_event(ak.Job.Event.LASERJOB_STARTED)
 
-	def _append_data_to_file(self, data):
-		if self._analyticsOn:
+	# -------- WRITER THREAD (queue --> analytics file) ----------------------------------------------------------------
+	def _write_queue_to_analytics_file(self):
+		while self._analytics_enabled:
 			try:
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: bring the try/catch to the outer most level

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 2]
```diff
@@ -42,900 +40,635 @@ def existing_analyticsHandler():
...
-				if not os.path.isfile(self._jsonfile):
-					self._init_jsonfile()
-				dataString = json.dumps(data, sort_keys=False) + '\n'
-				with open(self._jsonfile, 'a') as f:
-					f.write(dataString)
+				if not os.path.isfile(self._json_file):
+					self._init_json_file()
+
+				with open(self._json_file, 'a') as f:
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: first check, then open

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 3]
```diff
@@ -42,900 +40,635 @@ def existing_analyticsHandler():
...
+					while not self._analytics_queue.empty():
+						data = self._analytics_queue.get()
+						data_string = json.dumps(data, sort_keys=False) + '\n'
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: another try except for json.dumps

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 4]
```diff
@@ -42,900 +40,635 @@ def existing_analyticsHandler():
 
 
 class AnalyticsHandler(object):
-	DISK_SPACE_TIMER = 3.0
-	IP_ADDRESSES_TIMER = 15.0
-	SELF_CHECK_TIMER = 20.0
-	INTERNET_CONNECTION_TIMER = 25.0
-	MAINTENANCE_TIMER = 5.0
-	SELF_CHECK_USER_AGENT = 'MrBeamPlugin self check'
+	QUEUE_MAXSIZE = 100
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: 1000

**Review by andywerner — APPROVED** [review]

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 4 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: done

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 3 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: Done

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 2 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: done

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 1 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: done

**Pull request merged**